app/src/main/java/de/tum/in/tumcampusapp/database/repository/LocalRepositoryImpl.kt

```diff
@@ -0,0 +1,33 @@
+package de.tum.`in`.tumcampusapp.database.repository
+
+import android.content.Context
+import de.tum.`in`.tumcampusapp.database.TcaDb
+import de.tum.`in`.tumcampusapp.models.cafeteria.Cafeteria
+import io.reactivex.Flowable
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+
+class LocalRepositoryImpl(val db: TcaDb, private val executor: Executor) : LocalRepository {
```

Comment:
Why not make this `object` instead of `class` - then you have a singleton the kotlin way?

Reply:
listen to @kordianbruck

Reply:
good point, but we'd have to pass at least the db to every method